src/watch_backend/RetrieveLocationBasedWeatherInfo.py

```diff
@@ -0,0 +1,85 @@
+import urllib.request, json
+from urllib.error import URLError
+
+from src.watch_constants.GenericConstants import *
+
+
+class RetrieveLocationBasedWeatherInfo:
+    def __init__(self):
+        # setting these values with some defaults here - you are more than welcome to modify it to your interests.
+        self.desc_cond: str = "Yeehaw"
+        self.general_temperature: float = 420.00
+        self.icon_id: str = "01d"
+
+        # location will default to Bentonville City Square
+        self.lat: float = 36.3725
+        self.lon: float = -94.2106
+        self.country: str = "US"
+
+    '''
+    This method hits a ip-based geo-location service to determine the geographical coordinates of your location which 
+    would be used in getting the weather information for your area from the weather service.
+    @:returns - latitude, longitude
+    '''
+    def get_location(self) -> (float, float):
+        try:
+            with urllib.request.urlopen(ip_api_url) as url_request:
+                url_response = url_request.read().decode('utf-8')
+                ip_data = json.loads(url_response)
+            url_request.close()
+            # set the latitude & longitude
+            self.lat, self.lon = ip_data['lat'], ip_data['lon']
+        except (ConnectionError, URLError):
+            print("Connection Error... Using Default Location!")
+        return self.lat, self.lon
+
+    '''
+    This method takes geographical coordinates & calls the open-weather api to determine the weather conditions for a given
+    set of geo-coordinates.
+    @:param - latitude, longitude
+    @:return - current_temperature, weather_condition, icon_id
+    '''
+    def get_weather(self, lat, lon) -> (str, float, str):
+        weather_url = '{}lat={}&lon={}&appid={}'.format(base_weather_url, lat, lon, owm_key)
+        try:
+            api_response = urllib.request.urlopen(weather_url)
+            response_query = api_response.read().decode('utf-8')
+            response_data = json.loads(response_query)
+            api_response.close()
```

Comment:
think you can do something similair to 'with open(file)' on this connection, on error it may not close here